keps/sig-release/20190118-creating-packages.md

```diff
@@ -0,0 +1,417 @@
+     ok `k/k` (a version like in `bazel-genfiles/build/version`)
+  1. do a build (`build/run.sh make $COMPONENT` or such) or download all needed
+     artefacts from from a staging bucket
+  1. run the tool to prepare packages for upload
+  1. upload the packages
+
+
+[anago]: https://github.com/kubernetes/release/blob/master/anago
+
+<!--
+### User Stories [optional]
+
+Detail the things that people will be able to do if this KEP is implemented.
+Include as much detail as possible so that people can understand the "how" of the system.
+The goal here is to make this feel real for users without getting bogged down.
+
+#### Story 1
+
+#### Story 2
+
+### Implementation Details/Notes/Constraints [optional]
+
+What are the caveats to the implementation?
+What are some important details that didn't come across above.
+Go in to as much detail as necessary here.
+This might be a good place to talk about core concepts and how they releate.
+-->
+
+### Risks and Mitigations
+
+- *Risk*: creating another way on how to build packages  
+  *Mitigations*:  
+  - work with the current builder (@sumi) in google
+  - writing super extra great docs so this is the goto way on how to build
+- *Risk*: `nfpm` is not the right tool, it goes away, works differently, ...  
+  *Mitigations*:  
+  - we encapsulate the build of exactly one package, if things change, we can
+    change the internal implementation but keep the external
+    interface
+- *Risk*: We cannot build proper windows packages but community wants/needs them  
+  *Mitigations*:
+  - Talk to people in sig windows, esp. about packaging their components
+
+## Design Details
+
+### Test Plan
+
+- different components will have unit tests as much as possible
+- comparing output debs/rpms to what currently is released
+  - not sure how? by the files they deploy?
+
+### Graduation Criteria
+
+#### From `nil` to `k/k`, Introduce the new mechanism into `k/k`
+
+- Iterate here on the KEP
+- Do PoCs
+- Compare to packages generated with the current release tooling
+- See no significant difference in the generated packages
+- The new mechanism has means to prepare the packages in a way that they can be
+  published via the current release workflow
+
+#### Deprecate all other options to build packages
+
+Deprecation of other packaging mechanisms is done either by a note in the
+README or even in the source code to print a warning when building packages via
+deprecated tools.
+
+Before deprecation the following points should be implemented:
+
+- Make it easy for the release team to switch between the old and the new process
+- Make it easy for the release team to compare two (sets of) packages and
+  assess if they are sufficiently similar
+- Change the release (branch management & patch management) processes to use
+  the new mechanism by default
+
+#### Make this the only available way to build packages
+
+- X (TBD) releases have been cut with the new machanism
```

Comment:
2 might be enough.
possibly something for SIG release to vote on as i don't think we have an official deprecation policy for that.